### ISSUE TITLE
Changing indent_size

### ISSUE THREAD

**Issue by f4lco** [pinterest/ktlint#547]
Changing `indent_size` in `.editorconfig` appears to affect only a subset of lines, see examples below. Is this the desired behavior?

```
$ ./ktlint --version
0.34.2
```

Suppose we have below class, which is properly formatted according to Ktlint:

```kotlin
class MyClass(
    val x: Any,
    val y: Any
) {

    private fun doit() {
        if (true) {
            println(x)
        }
    }
}
```

and change the `.editorconfig` to switch from 4 spaces to 2:

```
[*.{kt,kts}]
indent_size = 2
```

and run Ktlint again:

```
./ktlint --verbose --debug --format MyClass.kt
[DEBUG] Discovered ruleset "standard"
[DEBUG] Discovered reporter "plain"
[DEBUG] Discovered reporter "json"
[DEBUG] Discovered reporter "checkstyle"
[DEBUG] Initializing "plain" reporter with {verbose=true, color=false}
[DEBUG] Checking /Users/redacted/MyClass.kt
[DEBUG] Discovered .editorconfig (/Users/redacted) {indent_size=2}
[DEBUG] 1606ms / 1 file(s) / 0 error(s)
```

We see that only the member properties changed, and the method remained untouched:

```diff
diff --git a/MyClass.kt b/MyClass.kt
index 8f70718..a0530d1 100644
--- a/MyClass.kt
+++ b/MyClass.kt
@@ -1,7 +1,7 @@

 class MyClass(
-    val x: Any,
-    val y: Any
+  val x: Any,
+  val y: Any
 ) {

     private fun doit() {
```

However, I expected Ktlint to also reduce the indentation of the method declaration, and of the contained statements. Am I missing something? After playing around more, I couldn't get Ktlint to move the method declaration at all. For example, after removing the whitespace in front of `private`, and then formatting again, the format stays the same.

**Comment by shashachu** [comment]
@f4lco can you try running with the `--experimental` flag? this will enable the experimental indentation rule, which does better autoformatting.

**Comment by f4lco** [comment]
@shashachu yes, `--experimental` produces the expected outcome:

```diff
diff --git a/MyClass.kt b/MyClass.kt
index 8f70718..a2d4b60 100644
--- a/MyClass.kt
+++ b/MyClass.kt
@@ -1,12 +1,12 @@

 class MyClass(
-    val x: Any,
-    val y: Any
+  val x: Any,
+  val y: Any
 ) {

-    private fun doit() {
-        if (true) {
-            println(x)
-        }
+  private fun doit() {
+    if (true) {
+      println(x)
     }
+  }
 }
```
So I guess there is nothing left to do than wait for the experimental indentation rule to stabilize. Right now, I can use the flag. Thanks a lot!
